llpc/lower/llpcSpirvLowerMath.cpp

```diff
@@ -241,6 +242,63 @@ Function *SpirvLowerMathConstFolding::getEntryPoint() {
 }
 
 #undef DEBUG_TYPE // DEBUG_TYPE_CONST_FOLDING
+#define DEBUG_TYPE DEBUG_TYPE_PRECISION
+
+// =====================================================================================================================
+// Run precision (fast math flag) adjustment SPIR-V lowering pass on the specified LLVM module.
+//
+// @param [in/out] module : LLVM module to be run on (empty on entry)
+// @param [in/out] analysisManager : Analysis manager to use for this transformation
+PreservedAnalyses SpirvLowerMathPrecision::run(Module &module, ModuleAnalysisManager &analysisManager) {
+  runImpl(module);
+  return PreservedAnalyses::none();
```

Comment:
Since you already track if something changed in `runImpl`, you could use this result to avoid recomputing analyses if `runImpl` did nothing?

Reply:
Good point! Done.